src/ir/comp.rs

```diff
@@ -1473,8 +1473,27 @@ impl DotAttributes for CompInfo {
 impl IsOpaque for CompInfo {
     type Extra = ();
 
-    fn is_opaque(&self, _: &BindgenContext, _: &()) -> bool {
-        self.has_non_type_template_params
+    fn is_opaque(&self, ctx: &BindgenContext, _: &()) -> bool {
+        //Early return to avoid extra computation
```

Comment:
Nitpick: space between `//` and `Early`, and a period at the end of the sentence please :)